lib/src/Uncertainty/Bayesian/openturns/CalibrationResult.hxx

```diff
@@ -134,6 +135,17 @@ private:
   Sample outputAtPriorMean_;
   Sample outputAtPosteriorMean_;
 
+  /* Bayesian */
+  Bool bayesian_;
```

Comment:
I found an example of backward compatibility code at [GeneralLinearModelResult.cxx line 187](https://github.com/openturns/openturns/blob/defba869638ddf25996ed390cfac3c202999d66e/lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelResult.cxx#L187), but I do not know which study version (e.g. 102100) I should write. Where does this number come from?

Reply:
in your case you dont need it

instead, in the load method, you can check if its a new file by checking the presence of the attribute, and load it only then:
```
if (adv.hasAttribute("bayesian_"))
  load("bayesian_", bayesian_)
```
in the save method nothing changes, you just save that attribute

Reply:
Ok, this is much simpler. To detect if the object is bayesian, I changed the code as follows: if the `parameterPrior_` distribution has a covariance matrix for which the first entry (0,0) is strictly less that MaxScalar, then the object is Bayesian. This is because the least squares methods use a MaxScalar prior covariance matrix.

Do you agree on this way?